src/CMakeLists.txt

```diff
@@ -58,7 +58,19 @@ if(NOT SIMDJSON_SANITIZE)
   else()
     add_test(
       NAME "avoid_abort"
-      COMMAND sh -c "${NM}  $<TARGET_FILE_NAME:simdjson> |  ${GREP} abort || exit 0  && exit 1"
+      # Under FreeBSD, the __cxa_guard_abort symbol may appear but it is fine.
+      # So we want to look for <space><possibly _>abort as a test.
+      COMMAND sh -c "${NM}  $<TARGET_FILE_NAME:simdjson> |  ${GREP} ' _*abort' || exit 0  && exit 1"
+      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+    )
+    add_test(
+      NAME "avoid_cout"
+      COMMAND sh -c "${NM}  $<TARGET_FILE_NAME:simdjson> |  ${GREP} ' _*cout' || exit 0  && exit 1"
+      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+    )
+    add_test(
+      NAME "avoid_printf"
+      COMMAND sh -c "${NM}  $<TARGET_FILE_NAME:simdjson> |  ${GREP} ' _*printf' || exit 0  && exit 1"
```

Comment:
It's not at all necessary, but is there a way to restrict this to stdout and stderr file descriptor references?

Reply:
I don't know how to do it. We are looking at the symbols, right? Here I can guess that if there is a symbol starting by printf or _prinft then we called "printf". 

Really this is just a heuristic to try to spot the problems early, but it is not all entirely safe. It is just meant to be helpful.

Reply:
Yeah, I really wasn't sure how, just commenting in the hopes it would trigger an "aha yeah we totally can, I think I know how" :)

fput*, fprintf(stdout), fflush(stdout), etc. wouldn't be caught by these heuristics anyway ... but printf is *definitely* a good thing to catfch. I know I've left debug printfs in more than once on accident!